core/communications/objectWorkQueue.go

```diff
@@ -0,0 +1,113 @@
+				}
+				lockIndex := common.HashStrings(meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+				common.ObjectLocks.Lock(lockIndex)
+				var notificationsInfo []common.NotificationInfo
+
+				// if i.NotificationType == common.Update {
+				// 	if trace.IsLogging(logger.TRACE) {
+				// 		trace.Trace("Lily - prepare update notifications for %s/%s/%s", meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+				// 	}
+				// 	notificationsInfo, _ = PrepareObjectNotifications(*meta)
+				// } else if i.NotificationType == common.Delete {
+				// 	if trace.IsLogging(logger.TRACE) {
+				// 		trace.Trace("Lily - prepare delete notifications for %s/%s/%s", meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+				// 	}
+				// 	notificationsInfo, _ = PrepareDeleteNotifications(*meta)
+
+				// } else {
+
+				// }
+
+				switch i.NotificationAction {
+				case common.Update:
+					if i.NotificationType == common.TypeObject {
+						if trace.IsLogging(logger.TRACE) {
+							trace.Trace("Lily - prepare update notifications for %s/%s/%s", meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+						}
+						notificationsInfo, _ = PrepareObjectNotifications(*meta)
+					} else if i.NotificationType == common.TypeDestination {
+						if trace.IsLogging(logger.TRACE) {
+							trace.Trace("Lily - for object %s/%s/%s, prepare update destination notifications for destinations %s", meta.DestOrgID, meta.ObjectType, meta.ObjectID, i.Destinations)
+						}
+						if len(i.Destinations) > 0 {
+							notificationsInfo, _ = PrepareNotificationsForDestinations(*meta, i.Destinations, common.Update)
+						}
+					}
+
+				case common.Delete:
+					if i.NotificationType == common.TypeObject {
+						if trace.IsLogging(logger.TRACE) {
+							trace.Trace("Lily - prepare delete notifications for %s/%s/%s", meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+						}
+						notificationsInfo, _ = PrepareDeleteNotifications(*meta)
+					} else if i.NotificationType == common.TypeDestination {
+						if trace.IsLogging(logger.TRACE) {
+							trace.Trace("Lily - for object %s/%s/%s, prepare delete destination notifications for destinations %s", meta.DestOrgID, meta.ObjectType, meta.ObjectID, i.Destinations)
+						}
+						if len(i.Destinations) > 0 {
+							notificationsInfo, _ = PrepareNotificationsForDestinations(*meta, i.Destinations, common.Delete)
+						}
+					}
+				}
+
+				common.ObjectLocks.Unlock(lockIndex)
+				SendNotifications(notificationsInfo)
+
+				if trace.IsLogging(logger.TRACE) {
+					trace.Trace("Lily - Sent notifications for %s/%s/%s", meta.DestOrgID, meta.ObjectType, meta.ObjectID)
+				}
+
+			} else {
+				if trace.IsLogging(logger.TRACE) {
+					trace.Trace("Lily - Nothing from object Queue")
+				}
+
+			}
+		}
+	}
+
+}
+
+func (q *ObjectWorkQueue) Close() {
+	close(q.objectQueue)
+}
+
+func (q *ObjectWorkQueue) SendObjectToQueue(objectInQueue common.ObjectInQueue) {
+	go func() {
```

Comment:
If the channel hits the buffered limited, these go routines will pile up until the channel starts to empty out. Why did you decide to do this? I think it would be better to go with a larger channel buffer. A go thread probably takes up more memory than an ObjectInQueue object.